### DISCUSSION TITLE
HTTP Endpoint with POST

### DISCUSSION THREAD

**Discussion by fulganta** [elsa-workflows/elsa-core#1556] [Design & tech]
Hi,

In your tutorial https://medium.com/codex/part-6-of-building-workflow-driven-net-applications-with-elsa-2-9b3167c612dd
you access the workflow with a post command: 

```
POST https://localhost:5001/v1/workflows/{your-workflow-id}/execute' \
--header 'Content-Type: application/json' \ --data-raw '{"correlationId": "{your-document-id}
```

I would like to know how to do it with an activity HTTP Endpoint POST.

How to configure the HTTP Endpoint Activity to trigger a workflow and send input data to a custom activity such as the DocumentFile activity in the tutorial?

**Comment by sfmskywalker** [comment]
Hi @cfrelet,

If you follow [this guide](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-document-approval), you will see the HTTP Endpoint activity in action, as well as how to post JSON to it which is then used further in the workflow. Once you grasp the concepts shown there, you should be able to apply them to your particular use case.

If you have further questions still, feel free to continue the discussion here.

**Reply by fulganta** [reply]
Many thx for your quick feedback.
It seems to work fine!

I did what they say here and it works fine:

```
HTTP Endpoint
Now click the Start button and look for the HTTP Endpoint activity and select it. Configure it with the following settings:

Path: /v2/documents
Methods: POST
Read Content: true (checked)
Set Document Variable
Click the Done outcome button on the previous activity and look for the Set Variable activity and configure it as follows:

Variable Name: Document
Value: input.Body (JavaScript)
```

**Comment by Rathna314** [comment]
I am trying to execute elsa workflow designer HTTPEndpoint from another API. I need to send Json in that url. How do I do it, any ideas?

This is my controller code: 
var request = new HttpRequestMessage
                {
                    Method = HttpMethod.Get,
                    RequestUri = new Uri("https://example.com/api/resource"),
                    Headers =
                    {
                        { "User-Agent", "YourUserAgent" },
                        // Add other headers if necessary
                    },
                };

                using (var response = await _httpClient.SendAsync(request))
                {
                    response.EnsureSuccessStatusCode();
                    var content = await response.Content.ReadAsStringAsync();
                    return Ok(content);
                }
            }
